slick.core.ts

```diff
@@ -5,13 +5,15 @@
  */
 
 (function ($) {
+  type TEventData = Event & JQuery.Event;
+
   /***
    * An event object for passing data to event handlers and letting them control propagation.
    * <p>This is pretty much identical to how W3C and jQuery implement events.</p>
    * @class EventData
    * @constructor
    */
-  function EventData() {
+  const EventData = (function (this: TEventData): void {
```

Comment:
does `function EventData(this: TEventData): void {` not work too?

Reply:
No I tried that, but if you can get it to work, I would prefer that over the function expression

Reply:
I need it to be a function expression to recast it with a constructor signature

Reply:
could this be an actual class?

Reply:
I understand the footprint of the change would be much higher

Reply:
Yeah, that is basically why I didn't use the actual class. 

Fundamentally this is the same and the type system understands what it is.